### PR TITLE
Add SVTR Chinese.benchmark model and Update CRNN benchmark model, and others

### PR DESCRIPTION
**Pull request by zhtmike**
1. CRNN Model improvement
Scene: 59.71% -> 60.45%
Web: 64.86% -> 65.95%
Document: 89.23% -> 97.68%

2. Seq2Seq Model Improvement
Scene: 55.39% -> 62.15%
Web: 61.90% -> 67.05%
Document: 97.05% -> 97.60%
FPS: 1493 -> 2160

2. Add SVTR Model
Scene: 65.93%
Web: 69.64%
Document: 98.01%

4. Chinese benchmark model supported vertical image recognition

5. Support filtering the samples when the label length is larger than the `max_label_len` during training

6. Add AMP convert during model export

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/rec/crnn/crnn_resnet34_ch.yaml [hunk 1]
```diff
@@ -23,34 +25,33 @@ model:
...
 postprocess:
   name: RecCTCLabelDecode
   character_dict_path: *character_dict_path
   use_space_char: *use_space_char
- 
+
 metric:
   name: RecMetric
   main_indicator: acc
+  lower: False
```
Comment (SamitHuang) [inline configs/rec/crnn/crnn_resnet34_ch.yaml:41]: How does it affect the results?

File: configs/rec/svtr/svtr_tiny.yaml [hunk 2]
```diff
@@ -57,9 +72,9 @@ metric:
   print_flag: False
 
 loss:
-  name: CTCLoss 
-  pred_seq_len: 25
-  max_label_len: *max_text_len  # this value should be smaller than pre_seq_len
+  name: CTCLoss
+  pred_seq_len: 25 # 100 / 4
```
Comment (SamitHuang) [inline configs/rec/svtr/svtr_tiny.yaml:76]: How can we confirm the optimal value for `pred_seq_len`?

File: configs/rec/svtr/svtr_tiny_ch.yaml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    dataset_root: path/to/data_lmdb_release/
+    data_dir: training/
+    label_file: null
+    sample_ratio: 1.0
+    shuffle: True
+    filter_max_len: True
+    max_text_len: *max_text_len
+    transform_pipeline:
+      - DecodeImage:
+          img_mode: BGR
```
Comment (SamitHuang) [inline configs/rec/svtr/svtr_tiny_ch.yaml:110]: Consider setting bgr_to_rgb: False, and bgr_to_rgb: False, which can has some influence on computing results but save some time.

File: mindocr/losses/rec_loss.py [hunk 4]
```diff
@@ -47,42 +42,22 @@ def construct(self, pred: Tensor, label: Tensor):
...
-        #T, bs, nc = logit.shape
-        #logit = ops.reshape(logit, (T*bs, nc))
         label_values = ops.reshape(label, (-1,))
 
         loss, _ = self.ctc_loss(logit, self.label_indices, label_values, self.sequence_length)
-
-        if self.reduction=='mean':
-            loss = loss.mean()
-
+        loss = self.get_loss(loss)
```
Comment (SamitHuang) [inline mindocr/losses/rec_loss.py:48]: is it the same as loss.mean if reduction == 'mean'?

**Review by SamitHuang — APPROVED** [review]
Nice work!

File: configs/rec/svtr/svtr_tiny.yaml (hunk 2 above)
Comment (zhtmike) [reply configs/rec/svtr/svtr_tiny.yaml:76]: it is based on the network output, cannot be tuned unless we modify the head

File: mindocr/losses/rec_loss.py (hunk 4 above)
Comment (zhtmike) [reply mindocr/losses/rec_loss.py:48]: If the mode is fp16, then self.get_loss will cast the input to be fp32 first and then calculate the mean, which prevent overflow during sum/mean calculation under fp16. So this is safer.

File: tools/export.py [hunk 5]
```diff
@@ -96,8 +97,13 @@ def check_args(args):
         type=str,
         default="",
         help='Directory to save the exported mindir file.')
+    parser.add_argument(
+        '--amp_level',
```
Comment (HaoyangLee) [inline tools/export.py:101]: If user input a yaml file path (the amp_level is set in yaml) to export mindir, he supposes all the args in yaml is loaded when exporting. However, we set default amp_level to be O0 here, which may be different in the one in user's yaml. This may cause misleading to user.
It's suggested to set --amp_level `required=True`. Add `choices=['O0', 'O1', 'O2', 'O3']`would be better. :)

File: mindocr/data/transforms/rec_transforms.py [hunk 6]
```diff
@@ -400,6 +400,30 @@ def __call__(self, data):
         return data
 
 
+class Rotate90IfVertical:
+    """Rotate the image by 90 degree when the height/width ratio is larger than the given threshold.
+    Note: It needs to be called before image resize."""
+    def __init__(self, threshold: float = 2.0, direction: str = 'anti_clockwise',  **kwargs):
```
Comment (HaoyangLee) [inline mindocr/data/transforms/rec_transforms.py:406]: The threshold of h/w of paddle and other part in mindocr is 1.5. Refer to: https://github.com/mindspore-lab/mindocr/blob/f399c7b1809a1df5634ddd9bdae00a6a882c5f9d/tools/infer/text/utils.py#L64

Comment (HaoyangLee) [inline mindocr/data/transforms/rec_transforms.py:406]:
```suggestion
    def __init__(self, threshold: float = 2.0, direction: str = 'counterclockwise',  **kwargs):
```

**Review by HaoyangLee — APPROVED** [review]
